R/PipeOpLearner.R

```diff
@@ -80,8 +80,10 @@ PipeOpLearner = R6Class("PipeOpLearner", inherit = PipeOp,
   public = list(
     initialize = function(learner, id = if (is.character(learner)) learner else learner$id, param_vals = list()) {
       private$.learner = as_learner(learner)$clone(deep = TRUE)  # FIXME: use `clone=TRUE` when mlr-org/mlr3#344 is fixed
-      task_type = mlr_reflections$task_types[private$.learner$task_type]$task
-      out_type = mlr_reflections$task_types[private$.learner$task_type]$prediction
+      task_types = copy(mlr_reflections$task_types)
+      setorder(task_types, package)
+      task_type = task_types[type == private$.learner$task_type]$task[1L]
```

Comment:
does
```r
task_types[type == private$.learner$task_type][order(package)[1], "task"]
```
also work here? (or something like this, I can't try it rn, I hope you get what I mean) if so it would be nice to avoid the `copy` and `setorder`.

also: please add a #fixme comment with a link to the relevant issue in mlr3spatiotemp and / or mlr3

also: I think you should use `get("type")` if you refer to variables *inside* the `data.table` because the static checker of cran will complain otherwise.

Reply:
good suggestions! Thanks! I dropped the copy and setorder and added a FIXME linking to mlr-org/mlr3#470